package/samplers/smac_sampler/sampler.py

```diff
@@ -0,0 +1,274 @@
+                min_samples_split=rf_min_samples_split,
+                min_samples_leaf=rf_min_samples_leaf,
+            )
+        else:
+            raise ValueError(f"Unknown Surrogate Model Type {model_type}")
+
+    def _get_acq_func(self, acq_func_type: str) -> AbstractAcquisitionFunction:
+        all_acq_func = {"ei": EI(), "ei_log": EI(log=True), "pi": PI(), "lcb": LCB()}
+        return all_acq_func[acq_func_type]
+
+    def _get_init_design(
+        self,
+        scenario: Scenario,
+        init_design_type: str,
+        n_configs: int | None = None,
+        n_configs_per_hyperparameter: int | None = 10,
+        max_ratio: float = 0.25,
+    ) -> AbstractInitialDesign:
+        if init_design_type == "sobol":
+            init_design = SobolInitialDesign(
+                scenario=scenario,
+                n_configs=n_configs,
+                n_configs_per_hyperparameter=n_configs_per_hyperparameter,
+                max_ratio=max_ratio,
+            )
+        elif init_design_type == "lhd":
+            init_design = LatinHypercubeInitialDesign(
+                scenario=scenario,
+                n_configs=n_configs,
+                n_configs_per_hyperparameter=n_configs_per_hyperparameter,
+                max_ratio=max_ratio,
+            )
+        elif init_design_type == "random":
+            init_design = RandomInitialDesign(
+                scenario=scenario,
+                n_configs=n_configs,
+                n_configs_per_hyperparameter=n_configs_per_hyperparameter,
+                max_ratio=max_ratio,
+            )
+        else:
+            raise NotImplementedError(f"Unknown Initial Design Type: {init_design_type}")
+        return init_design
+
+    def sample_relative(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> dict[str, float]:
+        trial_info: TrialInfo = self.smac.ask()
+        cfg = trial_info.config
+        self._runs_instance_seed_keys[get_config_hash(cfg)] = (
+            trial_info.instance,
+            trial_info.seed,
+        )
+        params = {}
+        for name, hp_value in cfg.items():
+            if name in self._hp_scale_value:
+                hp_value = self._integer_to_step_hp(
+                    integer_value=hp_value, scale_info=self._hp_scale_value[name]
+                )
+            params[name] = hp_value
+
+        return params
+
+    def after_trial(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        state: TrialState,
+        values: Sequence[float] | None,
+    ) -> None:
+        # Transform the trail info to smac
+        params = trial.params
+        cfg_params = {}
+        for name, hp_value in params.items():
+            if name in self._hp_scale_value:
+                hp_value = self._step_hp_to_intger(hp_value, scale_info=self._hp_scale_value[name])
+            cfg_params[name] = hp_value
+
+        # params to smac HP
+        y = np.asarray(values)
```

Comment:
In Optuna, we have the information of the directions to optimize in `study.directions` if needed!

```python
values_to_minimize = [v if d == optuna.study.StudyDirection.MINIMIZE else -v  for d, v in zip(study.directions, values)]
```